platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/AbstractJavaCompileSpecFactory.java

```diff
@@ -76,12 +77,19 @@ private T chooseSpecFromCompileOptions(File fallbackJavaHome) {
             return getCommandLineSpec(JavaExecutableUtils.resolveExecutable(forkExecutable));
         }
 
-        return getForkingSpec(fallbackJavaHome);
+        final int languageVersion;
+        if (toolchain != null) {
+            languageVersion = toolchain.getLanguageVersion().asInt();
+        } else {
+            languageVersion = JavaVersion.current().ordinal() + 1;
```

Comment:
💭 There might be a better way than `ordinal`. The issue is that `getMajorVersion()` returns a `String`, and we obviously want to have `int` here. This is basically the implementation of `getMajorVersion()` without the `String` conversion.